lib/macroni/Conversion/Kernel/KernelRewriters.cpp

```diff
@@ -28,7 +28,32 @@ bool is_container_of(macroni::MacroExpansion exp) {
 
 bool is_rcu_dereference(macroni::MacroExpansion exp) {
   return has_results_and_n_parameters(exp, 1) &&
-         has_name(exp, "rcu_dereference");
+         (has_name(exp, "rcu_dereference") ||
+          has_name(exp, "rcu_dereference_bh") ||
+          has_name(exp, "rcu_dereference_sched"));
+}
+
+bool is_rcu_dereference_check(macroni::MacroExpansion exp) {
+  return has_results_and_n_parameters(exp, 2) &&
+         (has_name(exp, "rcu_dereference_check") ||
+          has_name(exp, "rcu_dereference_bh_check") ||
+          has_name(exp, "rcu_dereference_sched_check") ||
+          has_name(exp, "rcu_dereference_protected"));
+}
+
+bool is_rcu_access_pointer(macroni::MacroExpansion exp) {
+  return has_results_and_n_parameters(exp, 1) &&
+         has_name(exp, "rcu_access_pointer");
+}
+
+bool is_rcu_assign_pointer(macroni::MacroExpansion exp) {
+  return has_results_and_n_parameters(exp, 2) &&
+         has_name(exp, "rcu_assign_pointer");
+}
+
+bool is_rcu_replace_pointer(macroni::MacroExpansion exp) {
+  return has_results_and_n_parameters(exp, 3) &&
+         has_name(exp, "rcu_replace_pointer");
```

Comment:
Reimplement the `has_name` chains using type lists (see how VAST does it for its types) - main advantages are
* You can re-use the query on type level
* You don't have magic strings here

Reply:
I might need your help with this one... Having a hard time grokking how I would type lists (or type switches?) here